src/snowflake/connector/connection.py

```diff
@@ -1990,3 +1990,21 @@ def _log_telemetry_imported_packages(self) -> None:
                     connection=self,
                 )
             )
+
+    def is_valid(self) -> bool:
+        """This function tries to answer the question: Is this connection still good for sending queries?
+        Attempts to validate the connections both on the TCP/IP and Session levels."""
+        logger.debug("validating connection and session")
+        if self.is_closed():
+            logger.debug("connection is already closed and not valid")
+            return False
+
+        try:
+            logger.debug("trying to heartbeat into the session to validate")
+            hb_result = self.rest._heartbeat()
+            session_valid = hb_result.get("success")
+            logger.debug("session still valid? %s", session_valid)
+            return session_valid
```

Comment:
So, `.get()` I think could return `None` by default and your type-hint says that is not a valid return type. Do either
```suggestion
            return bool(session_valid)
```
Or tell `.get()` to return `False` default value.

Reply:
i like the first version better :) thank you for the suggestion and the review!